raft/raft.go

```diff
@@ -573,6 +573,11 @@ func (r *raft) advance(rd Ready) {
 	if len(rd.Entries) > 0 {
 		e := rd.Entries[len(rd.Entries)-1]
 		r.raftLog.stableTo(e.Index, e.Term)
+		if r.prs.Progress[r.id] != nil {
+			r.logger.Infof("Updating progress: %v", e.Index)
+			r.prs.Progress[r.id].MaybeUpdate(e.Index)
+		}
+		r.maybeCommit() // TODO: Consider moving to stepLeader loop...
```

Comment:
This isn't thread(goroutine) safe, because the raft may do it concurrently.

Reply:
No need to call `r.maybeCommit()` if you follow [comment](https://github.com/etcd-io/etcd/pull/14406/files#r959161759) above